### PR TITLE
Average Ratings

### PR DESCRIPTION
**Pull request by MichaelJJames**
Added a function to work out average rating and return a star rating

### CODE REVIEW

File: code/ControllerExtension.php [hunk 1]
```diff
@@ -36,6 +36,39 @@ public function processReview($data, \Form $form){
         $this->owner->redirectBack();
 
     }
+    
+    public function averageRating() {
+    
+        if($rating = \Reviewable\Review::get()->filter('ObjectID', $this->owner->ID)){
```
Comment (camfindlay) [inline code/ControllerExtension.php:42]: change variable to "$reviews" to be clearer you returning a set of reviews.

File: code/ControllerExtension.php [hunk 2]
```diff
@@ -36,6 +36,39 @@ public function processReview($data, \Form $form){
...
+
+            $countRating = $rating->Count();
+            $ratingTotal = 0;
+
+            foreach($rating as $review) {
```
Comment (camfindlay) [inline code/ControllerExtension.php:47]: change $rating to $reviews as per above.

File: code/ControllerExtension.php [hunk 3]
```diff
@@ -36,6 +36,39 @@ public function processReview($data, \Form $form){
...
+
+                $ratingTotal += $review->Rating;
+            
+            }
+
+            if(round($ratingTotal / $countRating) == 0) {
```
Comment (camfindlay) [inline code/ControllerExtension.php:53]: can this be done with a switch, that might be more efficient.

File: code/ControllerExtension.php [hunk 4]
```diff
@@ -36,6 +36,39 @@ public function processReview($data, \Form $form){
...
+                return '<img src="/reviewable/images/star-off.png" alt="1" title="bad"> <img src="/reviewable/images/star-off.png" alt="2" title="poor"> <img src="/reviewable/images/star-off.png" alt="3" title="regular"> <img src="/reviewable/images/star-off.png" alt="4" title="good"> <img src="/reviewable/images/star-off.png" alt="5" title="gorgeous">';
```
Comment (camfindlay) [inline code/ControllerExtension.php:54]: Is there a way to do this without returning html directly? It's muddying the controller and view layers :)

**Review by camfindlay — COMMENTED** [review]
@MichaelJJames thanks for this :) I'm reviewing now.

**Comment by MichaelJJames** [comment]
Fixes in #2

**Pull request closed without merging**